js/build.js

```diff
@@ -20,22 +19,36 @@ Fliplet.Widget.instance('image-gallery', function(data) {
       onResize: function() {
         wall.fitWidth();
         wall.refresh();
+      },
+      onComplete: function () {
+        $container.addClass('freewall-ready');
       }
     });
 
     if (!Fliplet.Env.get('interact')) {
       $(WALL_SELECTOR + ' .brick img').click(function() {
         var $clickedBrick = $(this)[0].parentElement;
-        data.options = data.options || {}
-        data.options.index = $clickedBrick.index - 1
 
-        Fliplet.Navigate.previewImages(data);
+        data.options = data.options || {};
+        data.options.index = $clickedBrick.index - 1;
+
+        // Update remote image URLs to authenticated URLs
+        Promise.all(_.map(_.get(data, 'images'), function (image, i) {
+          if (!Fliplet.Media.isRemoteUrl(image.url)) {
+            return Promise.resolve();
```

Comment:
just `return` is enough

Reply:
But this is not inside a chained Promise, will `Promise.all()` accept a non-Promise object inside the array parameter?

Reply:
1. Turns out the widget already supported this (not sure why Inna's result came back negative)
1. This PR is now really just some shuffle around of code to make the rendering smoother